lib/cocoapods/installer/analyzer.rb

```diff
@@ -429,59 +433,33 @@ def generate_target(target_definition, pod_targets)
           end
         end
 
-        target.pod_targets = filter_pod_targets_for_target_definition(pod_targets, target_definition)
+        target.pod_targets = filter_pod_targets_for_target_definition(target_definition, pod_targets)
 
         target
       end
 
       # Returns a filtered list of pod targets that should or should not be part of the target definition. Pod targets
       # used by tests only are filtered.
       #
-      # @param [Array<PodTarget>] pod_targets
-      #        the array of pod targets to check against
-      #
       # @param [TargetDefinition] target_definition
       #        the target definition to use as the base for filtering
       #
-      # @return [Array<PodTarget>] the filtered list of pod targets.
-      #
-      def filter_pod_targets_for_target_definition(pod_targets, target_definition)
-        pod_targets.select do |pod_target|
-          pod_target.target_definitions.include?(target_definition) && !pod_target_test_only?(pod_target, pod_targets)
-        end
-      end
-
-      # Returns true if a pod target is only used by other pod targets as a test dependency and therefore should
-      # not be included as part of the aggregate target.
-      #
-      # @param [PodTarget] pod_target
-      #        the pod target being queried
-      #
       # @param [Array<PodTarget>] pod_targets
       #        the array of pod targets to check against
       #
-      # @return [Boolean] if the pod target is only referenced from test dependencies.
+      # @return [Array<PodTarget>] the filtered list of pod targets.
       #
-      def pod_target_test_only?(pod_target, pod_targets)
-        name = pod_target.name
-        key = @test_pod_target_key.new(name, pod_targets)
-        if @test_pod_target_analyzer_cache.key?(key)
-          return @test_pod_target_analyzer_cache[key]
-        end
-        source = pod_targets.any? do |pt|
-          pt.dependent_targets.map(&:name).include?(name)
-        end
-        test = pod_targets.any? do |pt|
-          pt.test_dependent_targets.reject { |dpt| dpt.name == pt.name }.map(&:name).include?(name)
+      def filter_pod_targets_for_target_definition(target_definition, pod_targets)
```

Comment:
This method now becomes very simple. A pod target is part of an aggregate only if:

1) It's part of its target definition
2) The resolver did not mark this as a test only pod target.